### DISCUSSION TITLE
How to install libgcc1-8 on kirkstone-l4t-r32.7.x

### DISCUSSION THREAD

**Discussion by chhg82** [OE4T/meta-tegra#1120] [Q&A] [answered]
Hi
I'm coming from the dunfell branch (r32.7.2) and want to switch to the kirkstone-l4t-r32.7.x.
For certain applications with nvcc I need gcc-8, libgcc-8 and libgcc1-8. With IMAGE_INSTALL:append I can only install gcc-8, libgcc-8 comes as dependency. But libgcc1-8 is missing.
For me I wouldn't need the gcc-11.3.0, would that help to install libgcc1-8 somehow?
Finally I need: /usr/libexec/gcc/aarch64-poky-linux/8.5.0/cc1, under 11.3.0 there is cc1 for this version.
Thanks

**Comment by madisongh** [comment, marked as answer]
By libgcc1-8, are you talking about shared libraries for the gcc runtime?  The gcc-for-nvcc recipes explicitly disable the shared runtime build to avoid conflicting with the normal toolchain runtime.  They're really intended for very limited use - the intent being that you should be building your apps with the normal (gcc 11) toolchain and only using the gcc 8 compiler indirectly via nvcc for compiling `.cu` files. 

If you really need to use an older toolchain globally, like we used to do with dunfell and earlier branches, you'll need to add in the recipes and configuration settings for that yourself.

As for the package providing the gcc 8 version of `cc1`, running `oe-pkgdata-util find-path "*/cc1"` shows it's the `cpp-8` package that provides it.  And for on-target development, you can install the `libgcc-8-dev` package for the static libgcc-8 runtime support libraries.

**Reply by chhg82** [reply]
Thank you very much, `cpp-8` did it.
I only need the gcc8 for the Matlab GPUCoder, which generates a binary on the target under `~/MATLAB/server/*` when the connection is established. Apparently this code is not gcc11 compatible yet.
